packages/webapp/src/containers/Filter/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+// In tempStateReducer > filterReducer
+export interface ReduxFilterEntity {
+  [filterKey: /* e.g. STATUS, LOCATION */ string]: FilterState;
```

Comment:
It might be nice for the type for `filterKey` to be passed as a generic so that if we need to call this type from more specific filter components we can specify the enum

```ts
export type ReduxFilterEntity<FilterKey extends string = string> = Record<FilterKey, FilterState>;
```

Reply:
I can't read generics yet, but I'll give this a close study, thank you 😄

Reply:
Ah okay, so we can call it like

```ts
  animalsFilter: ReduxFilterEntity<AnimalsFilterKeys>
 ```
 
Nice! 👍